### PR TITLE
Fix mesh display in tutorial

### PR DESCRIPTION
**Pull request by drammock**
This should fix the mesh display color bug that has been in `tutorials/forward/50_background_freesurfer_mne.py` since the 1.0 release (last known good was 0.24).  Unfortunately we probably won't be able to tell because that tutorial is bricking Circle in other ways right now:

```
generating gallery for auto_tutorials/forward... [ 75%] 50_background_freesurfer_mne.py
Clean 1636.6 s : 1.73 GB
qt.qpa.xcb: QXcbConnection: XCB error: 128 (Unknown), sequence: 15178, resource id: 2098234, major code: 130 (Unknown), minor code: 2
```

...and I haven't been able to pinpoint the cause of that yet.

### CODE REVIEW

File: mne/viz/backends/_pyvista.py [hunk 1]
```diff
@@ -196,7 +196,10 @@ def __init__(self, fig=None, size=(600, 600), bgcolor='black',
         self.antialias = _get_3d_option('antialias')
         self.depth_peeling = _get_3d_option('depth_peeling')
         # smooth_shading=True fails on MacOS CIs
-        self.smooth_shading = _get_3d_option('smooth_shading')
+        if sys.platform == 'darwin':
+            self.smooth_shading = _get_3d_option('smooth_shading')
+        else:
+            self.smooth_shading = smooth_shading
```
Comment (larsoner) [inline mne/viz/backends/_pyvista.py:202]: It seems like we should have `smooth_shading=None` in the signature, then here:
```
if smooth_shading is None:
    smooth_shading = _get_3d_option('smooth_shading')
```
and then if CIs have a problem with smooth shading we should for them set `MNE_3D_OPTION_SMOOTH_SHADING=false` or whatever.

Comment (drammock) [reply mne/viz/backends/_pyvista.py:202]: Ok. smooth_shading defaults to True in several class/method signatures, do you want them all to be None, or just here?

Comment (larsoner) [reply mne/viz/backends/_pyvista.py:202]: If it's public then None makes sense. Basically if a user hasn't explicitly set smooth_shading, it should use the _get_3d_option under the hood

**Comment by drammock** [comment]
hmm... macOS CIs do not actually fail here.  Only failure is this on Windows 3.9 conda, in test `test_get_coef_multiclass_full[4-10-2]` and it's a ThreadExceptionWarning (unrelated?):

https://dev.azure.com/mne-tools/mne-python/_build/results?buildId=22143&view=logs&j=2bd7b19d-6351-5e7f-8417-63f327ab45bc&t=0c49e006-47ab-5675-a040-5d301cde7356&l=3354

But unfortunately the rendered tutorial is better, but not all the way fixed:

https://output.circle-artifacts.com/output/job/ea11ac33-7953-417e-ad83-aefc14a19f85/artifacts/0/dev/auto_tutorials/forward/50_background_freesurfer_mne.html#cortical-alignment-spherical

Locally it looks correct for me though.  More debugging necessary.

**Comment by larsoner** [comment]
> Locally it looks correct for me though. More debugging necessary.

My initial reaction to this was that it was probably a MESA bug. Locally I got good plots natively, but could replicate with xvfb-run, which uses MESA under the hood:
```console
$ MNE_3D_OPTION_MULTI_SAMPLES=1 MNE_SKIP_INSTANCE_ASSERTIONS=1 PATTERN=50_background_freesurfer xvfb-run make html_dev-pattern-memory
```
Once I had that, I could debug locally to figure it out. Things I changed:

1. Deal with what must be a MESA wireframe bug by setting `line_width > 1` and `render_lines_as_tubes=True`. It changes the rendering sufficiently that it renders properly.
2. Automatically set `smooth_shading=False` when `representation='wireframe'` inside `renderer.mesh()` because you never want it on for a wireframe (it won't render properly I don't think)
3. Cull to the +X tris so that you only get one set in the view (which is from the right), and change the distance to 150 (the spheres extend to 100 so this is a nice distance).
4. Fix a camera reset bug -- it wasn't affecting the plots but was a bug that I noticed. We shouldn't use the default `render=True` in our `reset_camera` call within `set_view` but rather call `plotter.reset_camera(render=False)` because we then immediately change some stuff about the view, then (optionally) call `update`. This should fix some jerky `Brain` behavior I've never been able to track down until now where opening a window causes the views to "flash" a bit on startup. It should also make things slightly faster because it avoids a render of a view that is never used.

You can see the fixed outputs here:

https://output.circle-artifacts.com/output/job/c42d266b-d50e-461c-a41c-f6cf7f8951f1/artifacts/0/dev/auto_tutorials/forward/50_background_freesurfer_mne.html#cortical-alignment-spherical

image

image

image

**Review by agramfort — APPROVED** [review]
🙏

**Pull request merged**